### PR TITLE
Add initial 'exec' implementation.

### LINKED ISSUE #8: Support Pod Exec
Needed for Kubernetes "Gold" client status

### LINKED ISSUE #9: Add a k8s web sockets library
We need a general library for handling Kubernetes protocol websockets connections.

This is needed for

Exec
Attach
Port-forward

### PR DESCRIPTION
**Pull request by brendandburns**
@arschles 

Fixes #8 
Fixes #9

### CODE REVIEW

File: exec-example.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import k8s = require('@kubernetes/typescript-node');
```
Comment (arschles) [inline exec-example.ts]: any idea what the `@kubernetes/` notation is? never seen that before

File: exec.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    public constructor(config: KubeConfig) {
+        this.handler = new WebSocketHandler(config);
+    }        
+
+    // TODO: make command an array and support multiple args
+    public exec(namespace: string, podName: string, containerName: string, command: string, stdout: stream.Writable | any, stderr: stream.Writable | any) {
+        var hasStdout = stdout != null;
+        var hasStderr = stderr != null;
+        var path = `/api/v1/namespaces/${namespace}/pods/${podName}/exec?command=${command}&container=${containerName}&container=nginx&stderr=${hasStderr}&stdout=${hasStdout}`;
```
Comment (arschles) [inline exec.ts]: does command need to be url escaped?

File: exec.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    // TODO: make command an array and support multiple args
+    public exec(namespace: string, podName: string, containerName: string, command: string, stdout: stream.Writable | any, stderr: stream.Writable | any) {
+        var hasStdout = stdout != null;
+        var hasStderr = stderr != null;
+        var path = `/api/v1/namespaces/${namespace}/pods/${podName}/exec?command=${command}&container=${containerName}&container=nginx&stderr=${hasStderr}&stdout=${hasStdout}`;        
+        this.handler.connect(path, null, (stream: number, buff: Buffer) => {
+            if (buff.length < 1) {
+                return;
+            }
+            if (stream == 1) {
```
Comment (arschles) [inline exec.ts]: it'd be nice to use the [`process.stdout`](https://nodejs.org/api/process.html#process_process_stdout) and [`process.stderr`](https://nodejs.org/api/process.html#process_process_stderr) constants here to help readability

File: web-socket-handler.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        client.on('httpResponse', (resp) => {
+            console.log(resp);
+        });
+
+        // TODO: handle 'ws://' here.
+        var url = 'wss://' + this.config.getCurrentCluster().server.substr(8) + path;
+
+        client.connect(url, protocols);
+    }
+}
```
Comment (arschles) [inline web-socket-handler.ts]: same here RE newline

File: exec.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+                }
+                if (stderr) {
+                    stderr.end();
+                }
+            } else {
+                console.log("Unknown stream: " + stream);
+            }
+        });
+    }
+}
```
Comment (arschles) [inline exec.ts]: can you add a newline at the end of the file?

**Review by arschles — CHANGES_REQUESTED** [review]
a few nits

File: exec-example.ts (hunk 1 above)
Comment (brendandburns) [reply exec-example.ts]: Yeah, this is the organization notation from `npm.org`.

File: web-socket-handler.ts (hunk 4 above)
Comment (brendandburns) [reply web-socket-handler.ts]: done.

File: exec.ts (hunk 5 above)
Comment (brendandburns) [reply exec.ts]: done.

File: exec.ts (hunk 3 above)
Comment (brendandburns) [reply exec.ts]: These are Kubernetes constants. Added constants in the code and referenced them.

File: exec.ts (hunk 2 above)
Comment (brendandburns) [reply exec.ts]: Fixed.

**Comment by brendandburns** [comment]
Comments addressed, please re-check.

Thanks!

**Pull request merged**